src/rulesets/asyncapi/index.json

```diff
@@ -0,0 +1,465 @@
+{
+  "formats": [
+    "asyncapi2"
+  ],
+  "rules": {
+    "asyncapi2-channel-path-no-empty-parameter-declaration": {
+      "description": "Channel path should not have empty parameter substitution pattern.",
+      "recommended": true,
+      "type": "style",
+      "given": "$.channels.*~",
+      "then": {
+        "function": "pattern",
+        "functionOptions": {
+          "notMatch": "{}"
+        }
+      }
+    },
+    "asyncapi2-channel-path-no-query-nor-fragment": {
+      "description": "Channel path should not include a query (`?`) or a fragment (`#`) delimiter.",
+      "recommended": true,
+      "type": "style",
+      "given": "$.channels.*~",
+      "then": {
+        "function": "pattern",
+        "functionOptions": {
+          "notMatch": "[\\?#]"
+        }
+      },
+      "tags": [
+        "given"
+      ]
+    },
+    "asyncapi2-channel-path-no-trailing-slash": {
+      "description": "Channel path should not end with a slash.",
+      "recommended": true,
+      "type": "style",
+      "given": "$.channels.*~",
+      "then": {
+        "function": "pattern",
+        "functionOptions": {
+          "notMatch": ".+\\/$"
+        }
+      },
+      "tags": [
+        "given"
+      ]
+    },
+    "asyncapi2-headers-schema-type-object": {
+      "description": "Headers schema type should be `object`.",
+      "message": "Headers schema type should be `object` ({{error}}).",
+      "severity": 0,
```

Comment:
@philsturgeon Could you please check this severity (and the other ones that are defined) make sense?

Reply:
I forget what the numbers mean, I think we can use "warn" and "error" right @P0lip

Reply:
@philsturgeon Fixed